aspnetcore/fundamentals/middleware/extensibility-third-party-container/sample/Startup.cs

```diff
@@ -0,0 +1,87 @@
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using SimpleInjector;
+using SimpleInjector.Lifestyles;
+using MiddlewareExtensibilitySample.Data;
+using MiddlewareExtensibilitySample.Middleware;
+
+namespace MiddlewareExtensibilitySample
+{
+    public class Startup
+    {
+        private Container container = new Container();
+
+        #region snippet1
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddMvc();
+
+            // Replace the default middleware factory with the 
+            // SimpleInjectorMiddlewareFactory. Pass in the 
+            // Simple Injector container when the factory is
+            // instantiated.
+            services.AddTransient<IMiddlewareFactory>(_ =>
+            {
+                return new SimpleInjectorMiddlewareFactory(container);
+            });
+
+            // Wraps ASP.NET requests in an 
+            // SimpleInjector.Lifestyles.AsyncScopedLifestyle.
+            services.UseSimpleInjectorAspNetRequestScoping(container);
+
+            // Provide the AppDbContext from the Simple Injector
+            // container whenever it's requested from the default
+            // service container.
+            services.AddScoped<AppDbContext>(provider => container.GetInstance<AppDbContext>());
```

Comment:
Why do you need this?

Reply:
Can't DI from SI into the page. See: https://github.com/simpleinjector/SimpleInjector/issues/362